app/controllers/orgs/rosters_controller.rb

```diff
@@ -146,6 +174,58 @@ def download_roster
     # rubocop:enable Metrics/MethodLength
     # rubocop:enable Metrics/AbcSize
 
+    def select_google_classroom
+      @google_classroom_courses = fetch_all_google_classrooms
+    end
+
+    def search_google_classroom
+      courses_found = fetch_all_google_classrooms.select do |course|
+        course.name.downcase.include? params[:query].downcase
+      end
+
+      respond_to do |format|
+        format.html do
+          render partial: "orgs/rosters/google_classroom_collection",
+                 locals: { courses: courses_found }
+        end
+      end
+    end
+
+    def import_from_google_classroom
+      students = list_google_classroom_students(params[:course_id])
+      return if students.nil?
+
+      if students.blank?
+        flash[:warning] = "No students were found in your Google Classroom. Please add students and try again."
+        redirect_to roster_path(current_organization)
+      else
+        current_organization.update_attributes(google_course_id: params[:course_id])
+        add_google_classroom_students(students)
+      end
+    end
+
+    # rubocop:disable Metrics/AbcSize
+    def sync_google_classroom
+      return if current_organization.google_course_id.nil?
+
+      latest_students = list_google_classroom_students(current_organization.google_course_id)
+      latest_student_ids = latest_students.collect(&:user_id)
+      current_student_ids = current_roster.roster_entries.collect(&:google_user_id)
+
+      new_student_ids = latest_student_ids - current_student_ids
+      new_students = latest_students.select { |s| new_student_ids.include? s.user_id }
+
+      add_google_classroom_students(new_students)
+    end
+    # rubocop:enable Metrics/AbcSize
+
+    def unlink_google_classroom
+      current_organization.update_attributes!(google_course_id: nil)
```

Comment:
`update_attribute` skips validations. Use `update` instead to _not_ skip validations. 
See https://davidverhasselt.com/set-attributes-in-activerecord/ for details.
```suggestion
      current_organization.update!(google_course_id: nil)
```